[IMP] test_themes: whitelist carousel snippet classes in themes test

rusp-odoo opened the pull request:
The new blog posts, events, and appointments carousel snippets introduce
layout classes that are not anchored to a data-snippet attribute. The
themes test flags these as unexpected, so they must be explicitly
whitelisted to prevent false test failures.

Community PR - https://github.com/odoo/odoo/pull/238099
Enterprise PR - https://github.com/odoo/enterprise/pull/101478

task-[5090459](https://www.odoo.com/odoo/project/974/tasks/5090459)

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/design-themes/pull/1199.png)](https://mergebot.odoo.com/odoo/design-themes/pull/1199)

rushil-b-patel committed:
[IMP] test_themes: whitelist carousel snippet classes in themes test

The new blog posts, events, and appointments carousel snippets introduce
layout classes that are not anchored to a data-snippet attribute. The
themes test flags these as unexpected, so they must be explicitly
whitelisted to prevent false test failures.

task-5090459

loco-odoo commented on test_themes/tests/test_new_page_templates.py:
```diff
@@ -114,7 +114,9 @@
-    's_carousel_indicators_hidden',
+    's_carousel_indicators_hidden', 's_blog_posts_carousel', 's_events_carousel',
+    's_dynamic', 's_blog_posts_post', 's_blog_posts_effect_marley',
+    's_blog_posts_post_picture_size_default', 's_appointments_carousel', 's_events_event',
```
This actually indicates that few of the classes you used in your new snippets do not correctly follow the convention. I let you check the `test_render_applied_templates` test but let's take the example of `s_blog_posts_post`: if you drop your new blog carousel on the page, there are `s_blog_posts_post` elements that do not have a `.s_blog_posts` ancestor
